### PR TITLE
feat(ChatForm): add emoji picker

### PR DESCRIPTION
**Pull request by juliahermak**
Screenshot 2023-11-18 19 43 31
Screenshot 2023-11-18 19 43 48

### CODE REVIEW

**Comment by bludnic** [comment]
[UI: Add emoji and move attach to the input field (copied from iOS)](https://trello.com/c/x5HC9Uyj/426-ui-add-emoji-and-move-attach-to-the-input-field-copied-from-ios)

**Comment by bludnic** [comment]
@juliahermak Merge dev into current branch. Fix conflicts.

File: src/lib/display-mobile.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export function isMobile() {
+  const displayMobile = window.innerWidth < 600
+  return displayMobile
+}
```
Comment (bludnic) [inline src/lib/display-mobile.js:4]: Convert to .ts

File: src/components/AChat/AChatForm.vue [hunk 2]
```diff
@@ -127,14 +136,58 @@ export default {
       this.message = this.messageText
       this.focus()
     }
+    window.addEventListener('keydown', (event) => {
+      if (event.ctrlKey && event.shiftKey && event.code === 'KeyE') {
+        this.openElement()
+      }
+    })
```
Comment (bludnic) [inline src/components/AChat/AChatForm.vue:143]: Need to unsubscribe when component is unmounted

**Review by bludnic — CHANGES_REQUESTED** [review]

**Review by bludnic — APPROVED** [review]

File: src/lib/display-mobile.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export function isMobile() {
+  const displayMobile = window.innerWidth < 600
```
Comment (adamant-al) [inline src/lib/display-mobile.js]: window.innerWidth < 450 or /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)

File: src/components/Chat/ChatEmojis.vue [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  >
+    <template #activator="{ props }">
+      <v-icon class="chat-emojis__icon" icon="mdi-emoticon-outline" size="28" v-bind="props" />
+    </template>
+
+    <emoji-picker @emoji:select="getEmoji"></emoji-picker>
+  </v-menu>
+</template>
+<script>
+import emojiPicker from '@/components/EmojiPicker.vue'
```
Comment (bludnic) [inline src/components/Chat/ChatEmojis.vue:16]:
```suggestion
import EmojiPicker from '@/components/EmojiPicker.vue'
```

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/lib/display-mobile.ts [hunk 5]
```diff
@@ -1,4 +1,6 @@
 export function isMobile() {
-  const displayMobile = window.innerWidth < 600
+  const displayMobile =
+    window.innerWidth < 450 ||
+    /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)
```
Comment (bludnic) [inline src/lib/display-mobile.ts:4]: Use `import { detect } from 'detect-browser'`. See example in components/Chat/Chat.vue

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/lib/display-mobile.ts [hunk 6]
```diff
@@ -1,6 +1,8 @@
-export function isMobile() {
-  const displayMobile =
-    window.innerWidth < 450 ||
-    /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)
-  return displayMobile
+import { detect } from 'detect-browser'
```
Comment (adamant-al) [inline src/lib/display-mobile.ts:1]: @bludnic @PaulDremanovich 
Let's choose a single function to check `isMobile`.
See https://github.com/Adamant-im/adamant-im/pull/601

File: src/components/AChat/AChatForm.vue [hunk 7]
```diff
@@ -229,6 +229,7 @@ export default {
       this.message += '\n'
     },
     focus() {
+      console.log('focus', this.$refs.messageTextarea)
```
Comment (adamant-al) [inline src/components/AChat/AChatForm.vue:232]: Remove logging.

**Pull request merged**